client/post-editor/editor-ground-control/history-button.jsx

```diff
@@ -12,26 +12,47 @@ import { flow } from 'lodash';
  * Internal dependencies
  */
 import { openPostRevisionsDialog } from 'state/posts/revisions/actions';
+import { withAnalytics, recordTracksEvent } from 'state/analytics/actions';
+import EditorRevisionsDialog from 'post-editor/editor-revisions/dialog';
 
-const HistoryButton = ( { openDialog, translate } ) => (
-	<div className="editor-ground-control__history">
-		<button className="editor-ground-control__history-button button is-link" onClick={ openDialog }>
-			{ translate( 'History' ) }
-		</button>
-	</div>
-);
+class HistoryButton extends Component {
+	render() {
+		const { loadRevision, postId, siteId, translate } = this.props;
+		return (
+			<div className="editor-ground-control__history">
+				<button
+					className="editor-ground-control__history-button button is-link"
+					onClick={ this.props.selectHistory }
+				>
+					{ translate( 'History' ) }
+				</button>
+				<EditorRevisionsDialog loadRevision={ loadRevision } postId={ postId } siteId={ siteId } />
+			</div>
+		);
+	}
+}
 
 HistoryButton.propTypes = {
 	// connected to dispatch
-	openDialog: PropTypes.func.isRequired,
+	openPostRevisionsDialog: PropTypes.func.isRequired,
+	recordTracksEvent: PropTypes.func.isRequired,
 
 	// localize
 	translate: PropTypes.func,
 };
 
+const selectHistory = () => dispatch => {
+	dispatch(
+		withAnalytics(
+			recordTracksEvent( 'calypso_editor_history_button_click' ),
+			openPostRevisionsDialog()
```

Comment:
You pass the function reference, not what it returns, so... without `()`.

Reply:
Thanks @aidvu. Removing the `()`.

Reply:
testing it out without the `()` and it doesn't seem to work anymore.

Reply:
Ow, lemme check `withAnalytics`. :)

Reply:
Ow, I see. :) My bad there. `openPostRevisionsDialog` is an action creator.

Reply:
No problem! thanks!